babel/no-invalid-this does not recognize private properties

E.g. if index.mjs contains

```js
export class Foo extends HTMLElement {
  foo = this.ok();
  ok = () => {
    return this.ok;
  }
  #notOk = () => {
    return this.#notOk;
  }
}
```

then eslint with the babel plugin will complain about the second `this` but not the first

```
$@ ./node_modules/.bin/eslint  invalid.mjs

/home/fergal/nodejs/invalid.mjs
  7:12  error  Unexpected 'this'  babel/no-invalid-this

✖ 1 problem (1 error, 0 warnings)


```

Slightly better example

```js
export class Foo extends HTMLElement {
  foo = this.ok();
  #foo = this.ok();
  ok = () => {
    return this.ok;
  }
  #notOk = () => {
    return this.#notOk;
  }
}
```

because that complains about both of the privates and neither of the publics.

```
  3:10  error  Unexpected 'this'  babel/no-invalid-this
  8:12  error  Unexpected 'this'  babel/no-invalid-this
```

--- a/rules/no-invalid-this.js
+++ b/rules/no-invalid-this.js
@@ -5,17 +5,18 @@ const eslint = require('eslint');
 const noInvalidThisRule = new eslint.Linter().getRules().get('no-invalid-this');
 
 module.exports = ruleComposer.filterReports(
-    noInvalidThisRule,    
+    noInvalidThisRule,
     (problem, metadata) => {
         let inClassProperty = false;
         let node = problem.node;
 
         while (node) {
-            if (node.type === "ClassProperty") {
+            if (node.type === "ClassProperty" ||
+                node.type === "ClassPrivateProperty") {
                 inClassProperty = true;
                 return;
             }
-            
+
             node = node.parent;
         }
 

--- a/tests/rules/no-invalid-this.js
+++ b/tests/rules/no-invalid-this.js
@@ -602,6 +602,21 @@ const patterns = [
         valid: [NORMAL, USE_STRICT, IMPLIED_STRICT, MODULES],
         invalid: []
     },
+
+    // Class Private Instance Properties.
+    {
+        code: "class A {#a = this.b;};",
+        parserOptions: { ecmaVersion: 6 },
+        valid: [NORMAL, USE_STRICT, IMPLIED_STRICT, MODULES],
+        invalid: []
+    },
+
+    {
+        code: "class A {#a = () => {return this.b;};};",
+        parserOptions: { ecmaVersion: 6 },
+        valid: [NORMAL, USE_STRICT, IMPLIED_STRICT, MODULES],
+        invalid: []
+    },
 ];
 
 const ruleTester = new RuleTester();

